src/signed_data.rs

```diff
@@ -243,36 +245,95 @@ impl<'a> SubjectPublicKeyInfo<'a> {
     }
 }
 
-/// A signature algorithm.
-pub struct SignatureAlgorithm {
+/// A detail-less error when a signature is not valid.
+#[derive(Debug, Copy, Clone)]
+pub struct InvalidSignature;
```

Comment:
Style nits: I would move the `InvalidSignature` type definition below the trait definition, and move the `RingSignatureAlgorithm` definition and impl above the trait definition.